schema/me/conversation.js

```diff
@@ -173,12 +202,24 @@ export const ConversationFields = {
     type: connectionDefinitions({ nodeType: MessageType }).connectionType,
     description: "A connection for all messages in a single conversation",
     args: pageable(),
-    resolve: ({ messages }, options) => {
-      const impulseOptions = parseRelayOptions(options)
-      return connectionFromArraySlice(messages, options, {
-        arrayLength: messages.length,
-        sliceStart: impulseOptions.offset,
-      })
+    resolve: ({ id, from_email_address }, options, req, { rootValue: { accessToken } }) => {
+      const { page, size, offset } = parseRelayOptions(options)
+      const impulseParams = { page, size, conversation_id: id }
+      return gravity
+        .with(accessToken, { method: "POST" })("me/token", {
+          client_application_id: IMPULSE_APPLICATION_ID,
+        })
+        .then(data => {
+          return impulse
+            .with(data.token, { method: "GET" })(`message_details`, impulseParams)
+            .then(({ total_count, message_details }) => {
+              message_details["conversation_from_address"] = from_email_address
```

Comment:
Got it, thanks for the explanation. So we just need this field to figure out if the message came from the user essentially?